src/gui/map_tooltip.cpp

```diff
@@ -792,82 +792,300 @@ void religion_map_tt_box(sys::state& state, text::columnar_layout& contents, dco
+		}
+
+		text::close_layout_box(contents, box);
+	}
+}
+
+void income_map_tt_box(sys::state& state, text::columnar_layout& contents, dcon::province_id prov) {
+	auto fat = dcon::fatten(state.world, prov);
+	country_name_box(state, contents, prov);
+
+	if(prov.value < state.province_definitions.first_sea_province.value) {
+		auto box = text::open_layout_box(contents);
+		float savings = 0.f;
+		for(auto pop : fat.get_pop_location()) {
+			savings += pop.get_pop().get_savings();
+		}
+		text::localised_single_sub_box(state, contents, box, std::string_view("mtt_total_income"), text::variable_type::prov, prov);
+		text::add_to_layout_box(state, contents, box, text::prettify_currency(savings), text::text_color::yellow);
+		text::close_layout_box(contents, box);
+	}
+}
+
+void ideology_map_tt_box(sys::state& state, text::columnar_layout& contents, dcon::province_id prov) {
+	auto fat = dcon::fatten(state.world, prov);
+	country_name_box(state, contents, prov);
+
+	if(prov.value < state.province_definitions.first_sea_province.value) {
+		auto box = text::open_layout_box(contents);
+
+		text::localised_single_sub_box(state, contents, box, std::string_view("mtt_dominant_ideology"), text::variable_type::prov, prov);
+
+		std::vector<dcon::ideology_fat_id> ideologies;
+		float total_pops = state.world.province_get_demographics(prov, demographics::total);
+		state.world.for_each_ideology([&](dcon::ideology_id id) {
+			if(fat.get_demographics(demographics::to_key(state, id)) > 0) {
+				ideologies.push_back(dcon::fatten(state.world, id));
+			}
+		});
+		std::sort(ideologies.begin(), ideologies.end(), [&](auto a, auto b) {return fat.get_demographics(demographics::to_key(state, a.id)) > fat.get_demographics(demographics::to_key(state, b.id)); });
+		for(size_t i = 0; i < ideologies.size(); i++) {
+			text::add_line_break_to_layout_box(state, contents, box);
+			text::add_space_to_layout_box(state, contents, box);
+			text::add_to_layout_box(state, contents, box, ideologies[i].get_name(), text::text_color::yellow);
+			text::add_space_to_layout_box(state, contents, box);
+			text::add_to_layout_box(state, contents, box, std::string_view("("), text::text_color::white);
+			text::add_to_layout_box(state, contents, box, text::format_percentage(fat.get_demographics(demographics::to_key(state, ideologies[i].id)) / fat.get_demographics(demographics::total)), text::text_color::white);
+			text::add_to_layout_box(state, contents, box, std::string_view(")"), text::text_color::white);
+		}
+
+		text::close_layout_box(contents, box);
+	}
+}
+
+void con_map_tt_box(sys::state& state, text::columnar_layout& contents, dcon::province_id prov) {
+	auto fat = dcon::fatten(state.world, prov);
+	country_name_box(state, contents, prov);
+
+	if(prov.value < state.province_definitions.first_sea_province.value) {
+		float average_con = (state.world.province_get_demographics(prov, demographics::consciousness) / state.world.province_get_demographics(prov, demographics::total));
```

Comment:
you may want to add a check for 0 total population here. This can happen in some mods (for example, those with Antarctica), as well as from seeing incomplete data mid update